model_compression_toolkit/core/common/hessian/hessian_service.py

```diff
@@ -0,0 +1,120 @@
+    """
+    def __init__(self):
+        self.hessian_cfg_to_hessian_data = {}  # Dictionary to store Hessians by configuration and image list
+        self._hessian_configurations = []  # hessian_configurations
+        self.input_data = None  # input_data
+        self.graph = None  # float graph
+        self.fw_impl = None
+
+    def set_graph(self, graph: Graph):
+        """Set the float graph for the service."""
+        self.graph = graph
+
+    def set_fw_impl(self, fw_impl):
+        """Set the framework implementation for the service."""
+        self.fw_impl = fw_impl
+
+    def add_hessian_configurations(self, hessian_configurations: List[HessianConfig]):
+        """Extend the current list of Hessian configurations with additional configurations."""
+        self._hessian_configurations.extend(hessian_configurations)
+
+    def _set_hessian_configurations(self, hessian_configurations: List[HessianConfig]):
+        """Set the list of Hessian configurations."""
+        self._hessian_configurations = hessian_configurations
+
+    def clear_cache(self):
+        """Clear the cached Hessian data."""
+        self.hessian_cfg_to_hessian_data={}
+
+    def count_cache(self,
+                    hessian_cfg: HessianConfig=None) -> int:
+        """
+        Count the cached Hessian data.
+        If a specific configuration is provided, it counts for that configuration.
+        Otherwise, it counts for all configurations.
+        """
+        if hessian_cfg:
+            if hessian_cfg in self.hessian_cfg_to_hessian_data:
+                return len(self.hessian_cfg_to_hessian_data[hessian_cfg])
+            return 0
+        return sum([len(x.values()) for x in self.hessian_cfg_to_hessian_data.values()])
+
+    def compute(self,
+                hessian_cfg:HessianConfig,
+                input_images: List[Any]):
+        """
+        Compute the Hessian based on the provided configuration and input images.
+        Store the computed Hessian in the cache.
+        """
+        if len(hessian_cfg.nodes_names_for_hessian_computation) == 1:
+            # Only one compare point, nothing else to "weight"
+            hessian = {hessian_cfg.nodes_names_for_hessian_computation[0]: 1.0}
+        else:
+            fw_hessian_calculator = self.fw_impl.get_framwork_hessian_calculator(hessian_cfg)
+            hessian_calculator = fw_hessian_calculator(graph=self.graph,
+                                                       config=hessian_cfg,
+                                                       input_images=input_images,
+                                                       fw_impl=self.fw_impl)
+            hessian = hessian_calculator.compute()
+
+        if hessian_cfg in self.hessian_cfg_to_hessian_data:
+            self.hessian_cfg_to_hessian_data[hessian_cfg][id(input_images)] = hessian
+        else:
+            self.hessian_cfg_to_hessian_data[hessian_cfg] = {id(input_images): hessian}
+
+    def fetch_hessian(self,
```

Comment:
split to two functions - all and for single input image

Reply:
It's been replaced with a method to fetch a number of approximations.